modules/git/repo_compare.go

```diff
@@ -286,6 +286,20 @@ func (repo *Repository) GetPatch(base, head string, w io.Writer) error {
 	return err
 }
 
+// GetFilesChangedBetween returns a list of all files that have been changed between the given commits
+func (repo *Repository) GetFilesChangedBetween(base, head string) ([]string, error) {
+	stderr := new(bytes.Buffer)
+	output := &strings.Builder{}
+	err := NewCommand(repo.Ctx, "diff", "--name-only", base+".."+head).
+		Run(&RunOpts{
+			Timeout: -1,
+			Dir:     repo.Path,
+			Stdout:  output,
+			Stderr:  stderr,
+		})
+	return strings.Split(output.String(), "\n"), err
```

Comment:
Using `RunStdString` can be more simple. `Timeout` can be omitted to use default timeout.